key.js

```diff
@@ -1,18 +1,321 @@
+        isGameOver: false,
+        sound: true,
+        gameAudio: {
+            getGameAudio: () => new Audio("GameAudio/gameBackgroundMusic.mp3"),
+            getClickAudio: () => new Audio("GameAudio/buttonPress.wav"),
+            getKeyboardSounds: () => ({
+                getInvalidAudio: () => new Audio('GameAudio/invalid.flac'),
+                getGreenAudio: () => new Audio()(`GameAudio/greenSound2.mp3`),
+                getRedAudio: () => new Audio("GameAudio/MPOP.wav")
+            })
+        },
+    });
+    const gameState = getInitialGameState();
+    function startGame() {
+        gameState.isStart = true;
+        gameState.currentLevel = 0;
+        gameState.gameAudio.getClickAudio();
+    }
+    function gameOver() {
+        if ((!gameState.lives || !gameState.seconds) && !gameState.levelsInfo.currentLevel) {
+            resetGameState();
+        }
+        if ((!gameState.lives || !gameState.seconds) && gameState.levelsInfo.currentLevel > 0) {
+            resetGameLivesAndTimer();
+        }
+    }
+    function levelUp() {
+        // Level 0 -> 5:
+        const currentLevel = gameState.levelsInfo.currentLevel;
+        if (!gameState.levelsInfo.setInfoList[currentLevel] && gameState.lives > 0 && gameState.seconds > 0) {
+            resetGameLivesAndTimer();
+            gameState.levelsInfo.currentLevel++;
+        }
+    }
+    function resetGameLivesAndTimer() {
+        const initialGameState = getInitialGameState();
+        gameState.lives = initialGameState.lives;
+        gameState.seconds = initialGameState.seconds;
+    }
+    function resetGameState() {
+        gameState = getInitialGameState();
+    }
+    function getGameLife() {
+        const gameLife = gameState.lives;
+        return gameLife;
+    }
+    function endGame() {
+        resetGameState();
+    }
+    function gameMusicPlay() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.currentTime = 0;
+        backgroundMusic.play();
+    }
+    function gameMusicStop() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.pause();
+    }
+    function muteAudio() {
+        gameState.sound = false;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => audio.pause());
+    }
+    function unMuteAudio() {
+        gameState.sound = true;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
```

Comment:
`keyboardSounds` should be a part of `gameState.gameAudio`.

```suggestion
            gameState.gameAudio.keyboardSounds.getInvalidAudio(),
            gameState.gameAudio.keyboardSounds.getGreenAudio(),
            gameState.gameAudio.keyboardSounds.getRedAudio()
```